en/guide/extending_csd.md

```diff
@@ -1,6 +1,121 @@
 # Extending Camera Streaming Daemon
 
-## Create a Custom Video Stream
+## Create a Custom Camera Device
+
+Camera-Streaming-Daemon has native support for auto detection of video cameras exported as Video4Linux(V4L2) devices. This covers most of regular Linux cameras, but there are some cameras that don't have V4L2 support or cases where user wants to set very specific parameters or do custom post-processing in video before exporting.
+
+For these use cases, it is possible to create a custom CameraDevice class, representing a single custom camera device.
+CameraServer class keeps a list of all created CameraComponents, each one holding a single available CameraDevice. These CameraDevice objects are used by CameraComponent class for camera control and ImageCapture and VideoStream classes to perform the image capture and streaming.
+
+[CameraDeviceGazebo](https://github.com/intel/camera-streaming-daemon/blob/master/src/CameraDeviceGazebo.cpp) is an example of a custom CameraDevice. We will explore it in order to learn all the steps necessary to add new custom CameraDevice in Camera-Streaming-Daemon.
+
+### CameraDevice
+
+CameraDevice is the abstract base class for all camera devices.
+
+```cpp
+class CameraDevice {
+```
+Every CameraDevice is identified by a unique string. For V4L2 class of camera devices, its the device node videox (For example video0 ). For Gazebo, its the string "gazebo".
+
+Based on how the camera frames can be read, there are two types of camera devices.
+- First is the camera device that supports gstreamer source element. The gst source element for such camera devices can be plugged in any gstreamer pipeline to implement the features like image capture, video streaming etc.
+- Second is the camera device that does not support gstreamer source element. Such type of class must implement the read function to return camera frame buffers. Gstreamer pipeline with appsrc element is created for such devices. The appsrc element is fed with frames returned by read function.
+
+### CameraParameters
+CameraParameters is the class to store and retrieve parameters and its value. Camera parameter is identified by a string and a corresponding integer ID. Parameter can be of any defined [types](http://mavlink.org/messages/common#MAV_PARAM_EXT_TYPE_UINT8). The parameter name, ID, type and its value(converted as string) is stored in the map data structure in the CameraParameters class. Few commonly used camera parameters are defined with name, ID and supported values. Nevertheless, developer can declare custom parameters as per their requirement.
+
+```cpp
+class CameraParameters {
+```
+#### 1. Extend CameraDevice Class
+To add support for new type of camera device in CSD, a custom class need to be derived from CameraDevice base class.
+
+Example: CameraDeviceGazebo is the class that extends CameraDevice class.
```

Comment:
I would omit this example - we can move the text up and everything is captured in the next example.

Reply:
10. Delete the line